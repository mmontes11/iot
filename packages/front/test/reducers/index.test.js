import deepFreeze from "deep-freeze";
import rootReducer, { isLoading, hasError } from "reducers";

jest.mock("reducers/auth", () => (state = false, { type }) => {
  switch (type) {
    case "ACTION":
      return "STATE";
    default:
      return state;
  }
});

describe("reducers/index", () => {
  it("combines other reducers", () => {
    expect(rootReducer).toMatchSnapshot();
  });
  const mockState = {
    auth: "AUTH",
    request: "REQUEST",
  };
  deepFreeze(mockState);
  it("has a default state", () => {
    expect(rootReducer(undefined, { type: "@init" })).toMatchSnapshot();
  });
  it("reduces nothing", () => {
    expect(rootReducer(mockState, { type: "WHATEVER" })).toMatchSnapshot();
  });
  it("reduces other actions", () => {
    expect(rootReducer(mockState, { type: "ACTION" })).toMatchSnapshot();
  });
  describe("isLoading", () => {
    it("indicates that the app is loading", () => {
      const state = {
        request: {
          pending: 1,
        },
      };
      expect(isLoading(state)).toBeTruthy();
    });
    it("indicates that the app is not loading", () => {
      const state = {
        request: {
          pending: 0,
        },
      };
      deepFreeze(state);
      expect(isLoading(state)).toBeFalsy();
    });
  });
  describe("hasError", () => {
    it("indicates that the app has an error", () => {
      const state = {
        request: {
          error: new Error(),
        },
      };
      expect(hasError(state)).toBeTruthy();
    });
    it("indicates that the app has not an error", () => {
      const state = {
        request: {
          error: null,
        },
      };
      deepFreeze(state);
      expect(hasError(state)).toBeFalsy();
    });
  });
});
