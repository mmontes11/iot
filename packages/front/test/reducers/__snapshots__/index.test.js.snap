// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers/index combines other reducers 1`] = `[Function]`;

exports[`reducers/index has a default state 1`] = `
Object {
  "app": Object {
    "isHamburgerMenuExpanded": false,
    "isMapDialogOpened": false,
    "showError": true,
  },
  "auth": false,
  "data": Object {
    "isLoading": false,
    "items": Array [],
    "things": Array [],
  },
  "filters": Object {
    "dateFilter": Object {
      "custom": Object {
        "endDate": null,
        "startDate": null,
      },
      "isCustomSelected": false,
      "timePeriod": Object {
        "isActive": false,
        "isDisabled": false,
        "isLoading": false,
        "items": Array [],
        "selectedItem": null,
      },
    },
    "items": Array [],
    "thingFilter": Object {
      "isActive": false,
      "isDisabled": false,
      "isLoading": false,
      "items": Array [],
      "selectedItem": null,
    },
    "type": Object {
      "isActive": false,
      "isDisabled": false,
      "items": Array [
        "thing",
        "date",
      ],
    },
  },
  "localization": Object {
    "languages": Array [
      "en",
      "es",
    ],
    "selectedLanguage": "en",
    "translations": Object {
      "Custom": "Custom",
      "Data": "Data",
      "Delete": "Delete",
      "End date": "End date",
      "Error": "Error",
      "Event stats": "Event stats",
      "Events:": "Events:",
      "Filters": "Filters",
      "Google maps": "Google maps",
      "Group by": "Group by",
      "IP address:": "IP address:",
      "IoT": "IoT",
      "Last observation:": "Last observation:",
      "Login": "Login",
      "Logout": "Logout",
      "MQTT topic:": "MQTT topic:",
      "Measurement stats": "Measurement stats",
      "Measurement type": "Measurement type",
      "Measurements:": "Measurements:",
      "Observation": "Observation",
      "Password": "Password",
      "Real time": "Real time",
      "Request failed": "Request failed",
      "Reset": "Reset",
      "Start date": "Start date",
      "Stats": "Stats",
      "Thing": "Thing",
      "Thing not found": "Thing not found",
      "Things": "Things",
      "Time": "Time",
      "Time period": "Time period",
      "Type": "Type",
      "Username": "Username",
      "avg": "Average",
      "avgByHour": "Average by hour",
      "date": "Date",
      "day": "Day",
      "door-closed": "Door closed",
      "door-opened": "Door opened",
      "en": "English",
      "es": "Spanish",
      "event": "Event",
      "hour": "Hour",
      "humidity": "Humidity",
      "humidity-indoor": "Humidity indoor",
      "humidity-outdoor": "Humidity outdoor",
      "luminosity": "Luminosity",
      "max": "Maximum",
      "maxByHour": "Maximum by hour",
      "measurement": "Mesasurement",
      "min": "Minimum",
      "minByHour": "Minimum by hour",
      "minute": "Minute",
      "month": "Month",
      "second": "Second",
      "stdDev": "Standard deviation",
      "stdDevByHour": "Standard deviation by hour",
      "temperature": "Temperature",
      "temperature-indoor": "Temperature indoor",
      "temperature-outdoor": "Temperature outdoor",
      "thing": "Thing",
      "total": "Total",
      "week": "Week",
      "year": "Year",
    },
  },
  "params": Object {
    "allIds": Array [],
    "byId": Object {},
    "isResetDisabled": true,
  },
  "request": Object {
    "error": null,
    "pending": 0,
    "statusCode": null,
  },
  "things": Object {
    "items": Array [],
    "selectedItem": null,
    "showNotFoundError": false,
  },
}
`;

exports[`reducers/index reduces nothing 1`] = `
Object {
  "app": Object {
    "isHamburgerMenuExpanded": false,
    "isMapDialogOpened": false,
    "showError": true,
  },
  "auth": "AUTH",
  "data": Object {
    "isLoading": false,
    "items": Array [],
    "things": Array [],
  },
  "filters": Object {
    "dateFilter": Object {
      "custom": Object {
        "endDate": null,
        "startDate": null,
      },
      "isCustomSelected": false,
      "timePeriod": Object {
        "isActive": false,
        "isDisabled": false,
        "isLoading": false,
        "items": Array [],
        "selectedItem": null,
      },
    },
    "items": Array [],
    "thingFilter": Object {
      "isActive": false,
      "isDisabled": false,
      "isLoading": false,
      "items": Array [],
      "selectedItem": null,
    },
    "type": Object {
      "isActive": false,
      "isDisabled": false,
      "items": Array [
        "thing",
        "date",
      ],
    },
  },
  "localization": Object {
    "languages": Array [
      "en",
      "es",
    ],
    "selectedLanguage": "en",
    "translations": Object {
      "Custom": "Custom",
      "Data": "Data",
      "Delete": "Delete",
      "End date": "End date",
      "Error": "Error",
      "Event stats": "Event stats",
      "Events:": "Events:",
      "Filters": "Filters",
      "Google maps": "Google maps",
      "Group by": "Group by",
      "IP address:": "IP address:",
      "IoT": "IoT",
      "Last observation:": "Last observation:",
      "Login": "Login",
      "Logout": "Logout",
      "MQTT topic:": "MQTT topic:",
      "Measurement stats": "Measurement stats",
      "Measurement type": "Measurement type",
      "Measurements:": "Measurements:",
      "Observation": "Observation",
      "Password": "Password",
      "Real time": "Real time",
      "Request failed": "Request failed",
      "Reset": "Reset",
      "Start date": "Start date",
      "Stats": "Stats",
      "Thing": "Thing",
      "Thing not found": "Thing not found",
      "Things": "Things",
      "Time": "Time",
      "Time period": "Time period",
      "Type": "Type",
      "Username": "Username",
      "avg": "Average",
      "avgByHour": "Average by hour",
      "date": "Date",
      "day": "Day",
      "door-closed": "Door closed",
      "door-opened": "Door opened",
      "en": "English",
      "es": "Spanish",
      "event": "Event",
      "hour": "Hour",
      "humidity": "Humidity",
      "humidity-indoor": "Humidity indoor",
      "humidity-outdoor": "Humidity outdoor",
      "luminosity": "Luminosity",
      "max": "Maximum",
      "maxByHour": "Maximum by hour",
      "measurement": "Mesasurement",
      "min": "Minimum",
      "minByHour": "Minimum by hour",
      "minute": "Minute",
      "month": "Month",
      "second": "Second",
      "stdDev": "Standard deviation",
      "stdDevByHour": "Standard deviation by hour",
      "temperature": "Temperature",
      "temperature-indoor": "Temperature indoor",
      "temperature-outdoor": "Temperature outdoor",
      "thing": "Thing",
      "total": "Total",
      "week": "Week",
      "year": "Year",
    },
  },
  "params": Object {
    "allIds": Array [],
    "byId": Object {},
    "isResetDisabled": true,
  },
  "request": "REQUEST",
  "things": Object {
    "items": Array [],
    "selectedItem": null,
    "showNotFoundError": false,
  },
}
`;

exports[`reducers/index reduces other actions 1`] = `
Object {
  "app": Object {
    "isHamburgerMenuExpanded": false,
    "isMapDialogOpened": false,
    "showError": true,
  },
  "auth": "STATE",
  "data": Object {
    "isLoading": false,
    "items": Array [],
    "things": Array [],
  },
  "filters": Object {
    "dateFilter": Object {
      "custom": Object {
        "endDate": null,
        "startDate": null,
      },
      "isCustomSelected": false,
      "timePeriod": Object {
        "isActive": false,
        "isDisabled": false,
        "isLoading": false,
        "items": Array [],
        "selectedItem": null,
      },
    },
    "items": Array [],
    "thingFilter": Object {
      "isActive": false,
      "isDisabled": false,
      "isLoading": false,
      "items": Array [],
      "selectedItem": null,
    },
    "type": Object {
      "isActive": false,
      "isDisabled": false,
      "items": Array [
        "thing",
        "date",
      ],
    },
  },
  "localization": Object {
    "languages": Array [
      "en",
      "es",
    ],
    "selectedLanguage": "en",
    "translations": Object {
      "Custom": "Custom",
      "Data": "Data",
      "Delete": "Delete",
      "End date": "End date",
      "Error": "Error",
      "Event stats": "Event stats",
      "Events:": "Events:",
      "Filters": "Filters",
      "Google maps": "Google maps",
      "Group by": "Group by",
      "IP address:": "IP address:",
      "IoT": "IoT",
      "Last observation:": "Last observation:",
      "Login": "Login",
      "Logout": "Logout",
      "MQTT topic:": "MQTT topic:",
      "Measurement stats": "Measurement stats",
      "Measurement type": "Measurement type",
      "Measurements:": "Measurements:",
      "Observation": "Observation",
      "Password": "Password",
      "Real time": "Real time",
      "Request failed": "Request failed",
      "Reset": "Reset",
      "Start date": "Start date",
      "Stats": "Stats",
      "Thing": "Thing",
      "Thing not found": "Thing not found",
      "Things": "Things",
      "Time": "Time",
      "Time period": "Time period",
      "Type": "Type",
      "Username": "Username",
      "avg": "Average",
      "avgByHour": "Average by hour",
      "date": "Date",
      "day": "Day",
      "door-closed": "Door closed",
      "door-opened": "Door opened",
      "en": "English",
      "es": "Spanish",
      "event": "Event",
      "hour": "Hour",
      "humidity": "Humidity",
      "humidity-indoor": "Humidity indoor",
      "humidity-outdoor": "Humidity outdoor",
      "luminosity": "Luminosity",
      "max": "Maximum",
      "maxByHour": "Maximum by hour",
      "measurement": "Mesasurement",
      "min": "Minimum",
      "minByHour": "Minimum by hour",
      "minute": "Minute",
      "month": "Month",
      "second": "Second",
      "stdDev": "Standard deviation",
      "stdDevByHour": "Standard deviation by hour",
      "temperature": "Temperature",
      "temperature-indoor": "Temperature indoor",
      "temperature-outdoor": "Temperature outdoor",
      "thing": "Thing",
      "total": "Total",
      "week": "Week",
      "year": "Year",
    },
  },
  "params": Object {
    "allIds": Array [],
    "byId": Object {},
    "isResetDisabled": true,
  },
  "request": "REQUEST",
  "things": Object {
    "items": Array [],
    "selectedItem": null,
    "showNotFoundError": false,
  },
}
`;
